scipy/optimize/_differentiable_functions.py

```diff
@@ -145,11 +145,17 @@ def update_grad():
                 self.g = grad_wrapped(self.x)
 
         elif grad in FD_METHODS:
+            lb, ub = _prepare_bounds(finite_diff_bounds, x0)
```

Comment:
```
C:\hostedtoolcache\windows\Python\3.7.9\x64\lib\site-packages\scipy\optimize\_numdiff.py:129: in _prepare_bounds
    lb = np.resize(lb, x0.shape)
E   AttributeError: 'list' object has no attribute 'shape'
```
Looks like `_prepare_bounds` doesn't like it if `x0` is a list, and apparently it's not guaranteed to be an array yet.

Reply:
changed